src/s3ql/backends/b2/b2_backend.py

```diff
@@ -527,10 +527,8 @@ def delete(self, key, force=False):
         else:
             file_ids = self._list_file_versions(key)
 
-        if not file_ids:
-            raise NoSuchObject(key)
-
-        self._delete_file_ids(file_ids, force)
+        if file_ids:
+            self._delete_file_ids(file_ids, force)
 
```

Comment:
`file_id, file_name = self._get_file_id_and_name(key)` can also raise `NoSuchObject` so when you want to swallow all 404s you need to also catch the `NoSuchObject` exception

Reply:
`NoSuchObject`, or `NoSuchKeyError` as in s3c.py ?
I'm not familiar with these APIs or this part of python.

Reply:
Nothing in s3c.py should affect the b2 backend, I believe

Reply:
Sorry, I mean, yes, I can capture NoSuchObject, but are there other Errors that I should also catch?
I'm not sure what errors can be raised in this situation that should be captured.
And I saw s2c would raise NoSuchKeyError, could that be raised too?

Reply:
You can completely ignore the s3c backend when working on the b2 backend.

I'm afraid the only way to find out what errors you should catch is to inspect the code that you are calling and look at what errors it may raise.